doc/ipython-notebooks/multiclass/KNN.ipynb

```diff
@@ -300,9 +394,11 @@
     "            dist = EuclideanDistance(feats, feats)\n",
     "            knn = KNN(k, dist, labels)\n",
     "            knn.set_store_model_features(True)\n",
-    "            if use_cover_tree:\n",
+    "            if knn_solver_type is KNN_COVER_TREE:\n",
     "                knn.set_knn_solver_type(KNN_COVER_TREE)\n",
-    "            else:\n",
+    "            elif knn_solver_type is KNN_KDTREE:\n",
+    "                knn.set_knn_solver_type(KNN_KDTREE)\n",
+    "            elif knn_solver_type is KNN_BRUTE:\n",
```

Comment:
BTW the cell below needs update: "Evaluate KNN with and without Cover Tree. This takes a few seconds:"

"Evaluate KNN using all possible solvers. This takes a few seconds"

It is also a bit weird that you go back and forth in dataset.
You should mention that you are going back to the initial dataset now